### PR TITLE
Adding Devspaces and automating devspaces provisioning

### PR DESCRIPTION
**Pull request by MazArslan**

### CODE REVIEW

File: argo-apps/tooling-app-of-apps/values.yaml [hunk 1]
```diff
@@ -26,5 +26,7 @@ applications:
      path: camelk-community  
    - name: tekton
      path: tekton  
-#    - name: devspaces
-#      path: devspaces  
+   - name: devspaces
```
Comment (agiertli) [inline argo-apps/tooling-app-of-apps/values.yaml]: Does this mean we have 2 separate application which deals with Dev Spaces? Is this needed? Can't we have a single app?

File: argo-apps/devspace-provisioning/templates/devworkspace-gitconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  name: devworkspace-gitconfig
+  namespace: {{.name}}-dev
+  annotations:
+    argocd.argoproj.io/sync-wave: "-1"    
+  labels:
+    app.kubernetes.io/defaultName: git-config-secret
+    app.kubernetes.io/part-of: devworkspace-operator
+    controller.devfile.io/watch-configmap: 'true'
+data:
+  gitconfig: "[filter \"lfs\"]\n    clean = git-lfs clean -- %f\n    smudge = git-lfs smudge -- %f\n    process = git-lfs filter-process\n    required = true\n\n[credential]\n    helper = store --file /.git-credentials/credentials\n\n[user]\n\tname = {{.name}}\n\temail = {{.name}}@che"
```
Comment (agiertli) [inline argo-apps/devspace-provisioning/templates/devworkspace-gitconfig.yaml]: We know the user's username and password, userN and openshift. Can we inject these credentials into the workspace and confirm that "git push" works without asking for credentials? We could first manually verify this procedure https://access.redhat.com/documentation/en-us/red_hat_openshift_dev_spaces/3.2/html-single/user_guide/index#using-git-credentials and if it works, we could gitopsify it. Thoughts?

File: install-lab/app-of-apps.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ spec:
   source:
     path: argo-apps/tooling-app-of-apps
     repoURL: https://github.com/agiertli/rhte-camelk.git
-    targetRevision: HEAD
+    targetRevision: maz
```
Comment (agiertli) [inline install-lab/app-of-apps.yaml]: I think we should roll this back to HEAD

File: argo-apps/tooling-app-of-apps/values.yaml [hunk 4]
```diff
@@ -26,5 +26,7 @@ applications:
...
+     path: devspaces
```
Comment (agiertli) [inline argo-apps/tooling-app-of-apps/values.yaml]: Also, dev spaces should be upgraded to 3.3 version which was released recently

File: argo-apps/devspace-provisioning/templates/devworkspace-gitconfig.yaml (hunk 2 above)
Comment (MazArslan) [reply argo-apps/devspace-provisioning/templates/devworkspace-gitconfig.yaml]: Done :)

File: argo-apps/tooling-app-of-apps/values.yaml (hunk 4 above)
Comment (MazArslan) [reply argo-apps/tooling-app-of-apps/values.yaml]: also done

File: argo-apps/tooling-app-of-apps/values.yaml [hunk 5]
```diff
@@ -14,17 +14,30 @@ applications:
...
    - name: tekton
      path: tekton  
-#    - name: devspaces
-#      path: devspaces  
+     destination: tooling
+   - name: devspaces-install
+     path: devspaces-install
+     destination: tooling
+   - name: devspace-seed
+     path: devspace-seed
```
Comment (agiertli) [inline argo-apps/tooling-app-of-apps/values.yaml]: Probably should be devspace**s**-seed , which might also result in renaming app directory itself.
And can we also add "destination: tooling" ?

**Review by agiertli — APPROVED** [review]
let's do this.

**Pull request merged**